tests/test_examples.py

```diff
@@ -1261,3 +1261,30 @@ def test_v1_0_arg_empty_prefix_separate_false() -> None:
     )
     assert "completed success" in stderr
     assert error_code == 0
+
+
+def test_scatter_output_filenames(tmpdir: py.path.local) -> None:
+    """Check that when a scatter step produces same named output the final output is renamed correctly."""
+    cwd = tmpdir.chdir()
+    rtc = RuntimeContext()
+    rtc.outdir = str(cwd)
+    factory = cwltool.factory.Factory(runtime_context=rtc)
+    output_names = ['output.txt', 'output.txt_2', 'output.txt_3']
+    try:
+        scatter_workflow = factory.make(get_data("tests/scatter_numbers.cwl"))
+        result = scatter_workflow(range=3)
+        assert(
+            'output' in result
+        )
+        locations = [element['location'] for element in result['output']]
+
+        assert(
+            locations[0].endswith('output.txt') and
+            locations[1].endswith('output.txt_2') and
+            locations[2].endswith('output.txt_3'),
+            "Locations {} do not end with {}".format(locations, output_names)
+        )
+    finally:
+        for filename in output_names:
+            if os.path.exists(filename):
+                os.remove(filename)
```

Comment:
I think pytest cleans up automatically if you use the `tmpdir` fixin?

https://docs.pytest.org/en/latest/tmpdir.html#the-tmpdir-fixture

Reply:
Perhaps. Other tests do `tmpdir.remove(ignore_errors=True)` e.g. https://github.com/common-workflow-language/cwltool/blob/main/tests/test_examples.py#L952

Reply:
I think that was a work around for a bug that was [fixed](https://github.com/pytest-dev/pytest/pull/6944) in Pytest 6.0.0 ; I'll make a separate PR to remove those